### PR TITLE
[WIP] Alpine docker

### PR DESCRIPTION
**Pull request by praneethmendu**
follows from [304](https://github.com/zkopru-network/zkopru/issues/304)

**context**
- we are currently installing python (along with other build tools )as a virtual package, using it to install truffle and deleting it immediately after to save space. But down the line we also need it for yarn install which is what keeps failing.

- there are some dockerfiles using debian among mostly alpine based images

**fixes**

- using a pattern that installs packages required for build tools temporarily. Seems to be standard practice in alpine dockerfiles

- installing python3 in this way where required, which truffle now needs

- all dockerfiles now use alpine

### CODE REVIEW

File: dockerfiles/Cli.dockerfile [hunk 1]
```diff
@@ -43,7 +25,16 @@ COPY ./packages/utils/package.json /proj/packages/utils/package.json
 COPY ./packages/zk-wizard/package.json /proj/packages/zk-wizard/package.json
 COPY ./yarn.lock /proj/yarn.lock
 
-RUN yarn install
+
+RUN npm install -g lerna
+# install build tools
+RUN apk add --no-cache --virtual .gyp \
```
Comment (JChanceHud) [inline dockerfiles/Cli.dockerfile:31]: Ideally we'd combine commands like this so that we have fewer docker layers. Not a huge concern though.

**Review by JChanceHud — CHANGES_REQUESTED** [review]

File: dockerfiles/Cli.dockerfile (hunk 1 above)
Comment (praneethmendu) [reply dockerfiles/Cli.dockerfile:31]: I am going through the dockerfiles to apply your feedback. So, in general, consecutive RUN layers can be grouped or are there some cases where they are better as separate layers ?

Comment (JChanceHud) [reply dockerfiles/Cli.dockerfile:28]: There are some cases, usually when you have something that doesn't always need to be done you will group it separately from things that _do_ always need to be done.

Like an install and a build are sometimes better as separate layers (though they're usually separated by a `COPY` step). Generally you'll know when things are better as separate layers. I've also run into weird cases where having one large `RUN` step produces a larger image than a few smaller `RUN` steps. It's a not a huge issue, just something that's good to know.

Comment (praneethmendu) [reply dockerfiles/Cli.dockerfile:28]: thank you for the detailed answer

**Comment by JChanceHud** [comment]
@praneethmendu Did you mean to close this PR? In the future you can force push to branches as well.

**Comment by praneethmendu** [comment]
@JChanceHud Yes that is what I meant to do. I did not know that answer before thank you

**Pull request closed without merging**